src/go/configgenerator/listener_generator.go

```diff
@@ -795,8 +800,15 @@ func makeBackendRoutingFilter(serviceInfo *sc.ServiceInfo) (*hcmpb.HttpFilter, e
 			}
 
 			if method.BackendInfo != nil {
-				newRule.PathPrefix = method.BackendInfo.Uri
+				newRule.PathPrefix = method.BackendInfo.Path
 			}
+
+			// If the path prefix is empty, then the filter is a no-op for this operation.
+			// It doesn't need to be in the filter config, skip it.
```

Comment:
If it is skipped,  will we reject this request as 404?

Reply:
Skipping the backend routing filter will skip rewriting the path. But we still configure a host rewrite in the router config.

So no, it will not result in a 404.

Reply:
ack